pkg/resources/virtual/virtual.go

```diff
@@ -23,6 +23,9 @@ func NewTransformBuilder(cache common.SummaryCache) *TransformBuilder {
 }
 
 // GetTransformFunc retrieves a TransformFunc for a given GVK. Currently only returns a transformFunc for defaultFields
-func (t *TransformBuilder) GetTransformFunc(_ schema.GroupVersionKind) cache.TransformFunc {
+func (t *TransformBuilder) GetTransformFunc(gvk schema.GroupVersionKind) cache.TransformFunc {
+	if gvk.Kind == "Event" {
```

Comment:
I would recommend checking full GVK here - events can also be implemented by other API groups and this would add transform functions for those types as well.

Reply:
Done, with an extra test.